packages/gamepads_web/lib/src/gamepad_detector.dart

```diff
@@ -0,0 +1,23 @@
+import 'dart:js_interop';
+
+import 'package:gamepads_platform_interface/api/gamepad_controller.dart';
+import 'package:gamepads_platform_interface/gamepads_platform_interface.dart';
+import 'package:web/web.dart';
+
+List<GamepadController> getGamepads(GamepadsPlatformInterface plugin) {
+  final controllers = <GamepadController>[];
+  return getGamepadList()
+    .map((gamepad) =>
+      GamepadController(
+        id: gamepad!.index.toString(),
+        name: gamepad.id,
+        plugin: plugin,
+      ),
+    );
+}
+
+List<Gamepad?> _getGamepadList() {
```

Comment:
Since you're removing all nulls, this doesn't need to return a list of nullables anymore.
```suggestion
List<Gamepad> _getGamepadList() {
```